arduino-ide-extension/src/common/protocol/library-service.ts

```diff
@@ -38,6 +49,86 @@ export interface LibraryService
   }): Promise<LibraryDependency[]>;
 }
 
+export interface LibrarySearch extends Searchable.Options {
+  readonly type?: LibrarySearch.Type;
+  readonly topic?: LibrarySearch.Topic;
+}
+export namespace LibrarySearch {
+  export const TypeLiterals = [
+    'All',
+    'Updatable',
+    'Installed',
+    'Arduino',
+    'Partner',
+    'Recommended',
+    'Contributed',
+    'Retired',
+  ] as const;
+  export type Type = typeof TypeLiterals[number];
+  export const TypeLabels: Record<Type, string> = {
+    All: All,
+    Updatable: Updatable,
+    Installed: nls.localize('arduino/libraryType/installed', 'Installed'),
+    Arduino: 'Arduino',
+    Partner: Partner,
+    Recommended: Recommended,
+    Contributed: Contributed,
+    Retired: Retired,
+  };
+  export const TopicLiterals = [
+    'All',
+    'Communication',
+    'Data Processing',
+    'Data Storage',
+    'Device Control',
+    'Display',
+    'Others',
+    'Sensors',
+    'Signal Input/Output',
+    'Timing',
+    'Uncategorized',
+  ] as const;
+  export type Topic = typeof TopicLiterals[number];
+  export const TopicLabels: Record<Topic, string> = {
+    All: All,
+    Communication: nls.localize(
+      'arduino/libraryTopic/communication',
+      'Communication'
+    ),
+    'Data Processing': nls.localize(
+      'arduino/libraryTopic/dataProcessing',
+      'Data Processing'
+    ),
+    'Data Storage': nls.localize(
+      'arduino/libraryTopic/dataStorage',
+      'Date Storage'
+    ),
+    'Device Control': nls.localize(
+      'arduino/libraryTopic/deviceControl',
+      'Device Control'
+    ),
+    Display: nls.localize('arduino/libraryTopic/display', 'Display'),
+    Others: nls.localize('arduino/libraryTopic/others', 'Others'),
```

Comment:
```suggestion
      'Data Storage'
    ),
    'Device Control': nls.localize(
      'arduino/libraryTopic/deviceControl',
      'Device Control'
    ),
    Display: nls.localize('arduino/libraryTopic/display', 'Display'),
    Others: nls.localize('arduino/libraryTopic/others', 'Other'),
```

Reference:

https://arduino.github.io/arduino-cli/dev/library-specification/#libraryproperties-file-format

Reply:
Fixed. Thank you!